compiler/testData/codegen/box/assert/jvm/initializerAssertionsEnabled.kt

```diff
@@ -0,0 +1,53 @@
+// TARGET_BACKEND: JVM
+// KOTLIN_CONFIGURATION_FLAGS: ASSERTIONS_MODE=jvm
+// WITH_RUNTIME
+
+// Assertions which run before the class initializer are always checked
+
+package initializerAssertionsEnabled
+
+class Checker {
+    fun test() = Baz.testAsserts()
+}
+
+open class Bar {
+    companion object {
+        val barAssertionThrown = try {
+            Baz().assertFalse()
+            false
+        } catch(error: java.lang.AssertionError) {
+            true
+        }
+    }
+}
+
+class Baz : Bar() {
+    fun assertFalse() = assert(false)
+
+    companion object {
+        val bazAssertionThrown = try {
+            Baz().assertFalse()
+            false
+        } catch(error: java.lang.AssertionError) {
+            true
+        }
+
+        fun testAsserts(): String {
+            if (!barAssertionThrown) return "Fail 1"
+            if (bazAssertionThrown) return "Fail 2"
+            return "OK"
+        }
+    }
+}
+
+class Dummy
+
+fun disableAssertions(): Checker {
+    val loader = Dummy::class.java.classLoader
+    loader.setPackageAssertionStatus("initializerAssertionsEnabled", false)
```

Comment:
Sorry for confusion by previous comment about 'initializerAssertionsDisable.kt' test , but seems it's best to rename this one to address name with ' // Assertions which run before the class initializer are always checked ' comment

Reply:
Right, I'll think of a better name, this one is misleading.

Reply:
Renamed the test to `testAssertionsEnabledBeforeClassInitializers`.